pkgs/by-name/op/open-adventure/package.nix

```
@@ -0,0 +1,81 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitLab,
+  python3Packages,
+  pkg-config,
+  libedit,
+  cppcheck,
+  coreutils,
+  asciidoctor,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "open-adventure";
+  version = "1.20";
+  src = fetchFromGitLab {
+    owner = "esr";
+    repo = "open-adventure";
+    tag = finalAttrs.version;
+    hash = "sha256-xbsMz99CNLhpM6BSJVcRzxPB6tUYfPy/3Z+8BKt8b1E=";
+  };
+
+  nativeBuildInputs = [
+    python3Packages.python
+    pkg-config
+    asciidoctor
+  ];
+
+  buildInputs = [
+    python3Packages.pyyaml
+    libedit
+  ];
+
+  doCheck = true;
+
+  nativeCheckInputs = [
+    python3Packages.pylint
+    cppcheck
+  ];
+
+  postPatch = ''
+    patchShebangs --build make_dungeon.py
+
+    # https://gitlab.com/esr/open-adventure/-/issues/70
+    substituteInPlace Makefile --replace-fail "--template " "--template="
+
+    substituteInPlace tests/tapview --replace-fail "/bin/echo" ${lib.getExe' coreutils "echo"}
+  '';
+
+  buildFlags = [
+    "advent.6"
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -vp "$out/bin" "$out/share/man/man6" "$out/share/applications/" "$out/share/icons/hicolor/scalable/apps"
+    install -m 555 ./advent $out/bin
+    install -m 444 ./advent.6 $out/share/man/man6
+    install -m 444 ./advent.desktop $out/share/applications
+    install -m 444 ./advent.svg $out/share/icons/hicolor/scalable/apps
```

A possible alternative that doesn't need mkdir. You don't need to do this if you don't want to.
```suggestion
    install -Dm555 ./advent -t $out/bin
    install -Dm444 ./advent.6 -t $out/share/man/man6
    install -Dm444 ./advent.desktop -t $out/share/applications
    install -Dm444 ./advent.svg -t $out/share/icons/hicolor/scalable/apps
```

Also, man-pages should ideally be installed via `installManPage` from the `installShellFiles` helper script collection. IMO this is not a hard requirement, so you can keep it as is.